pkg/k8s/service_cache.go

```diff
@@ -162,16 +217,36 @@ func (s *ServiceCache) DeleteService(k8sSvc *types.Service) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	oldService, serviceOK := s.services[svcID]
-	endpoints, _ := s.correlateEndpoints(svcID)
+	oldSvc, oldSvcOk := s.services[svcID]
+	oldEndpoints, _ := s.correlateEndpoints(svcID)
 	delete(s.services, svcID)
+	if svcID.k8sExternal {
+		// We will delete the external endpoints created in the service addition
+		delete(s.endpoints, svcID)
+
+		svcID.k8sExternal = false
+		oldNonExternalSvc, oldNonExternalSvcOk := s.services[svcID]
```

Comment:
`oldNonExternalSvcOK` will always be `false` because you deleted `svcID` from `s.services` on line 222.

Reply:
It will not, the `svcID` is a different key